src/ds/ds-timestamp.cpp

```diff
@@ -305,8 +305,7 @@ namespace librealsense
 
     rs2_time_t ds_timestamp_reader::get_frame_timestamp(const std::shared_ptr<frame_interface>& frame)
     {
-        std::lock_guard<std::recursive_mutex> lock(_mtx);
-        return _ts->get_time();
+        return time_service::get_time();
```

Comment:
What was the reason for the mutex originally?

Reply:
Hmm I don't know... but now that you mention it, I'll check if the mutex can be removed, too...

Reply:
I checked: the mutex is now between the frame counter changes (`frame_timestamp_reader::get_frame_counter()` when a frame is received and `::reset()` when a sensor is stopped). Before it included getting the current time (also on frame received). Maybe because frames can be received from parallel threads? Otherwise there'd be no reason for the mutex for counters, either...